python/tvm/script/special_stmt.py

```diff
@@ -142,21 +181,300 @@ def buffer_decl(
                 buffer_type,
                 span=span,
             )
-            self.context.update_symbol(self.node.lhs.id.name, buffer)
+            self.context.update_symbol(self.node.lhs.id.name, buffer, self.node)
             return buffer
 
         super().__init__(buffer_decl, def_symbol=True)
 
 
+@register
+class AllocBuffer(SpecialStmt):
+    """Special function alloc_buffer(shape, dtype, data, strides, elem_offset, scope, align,
+                                     offset_factor, buffer_type)
+
```

Comment:
parameters? (ditto to the rest)

Reply:
It is a little tricky that It's not the parameters for the class but for the registered stmts. So I don't know how to add docstrings properly